### ISSUE TITLE
Archive name on creation & Extract in other directory

### ISSUE THREAD

**Issue by ERROPiX** [Studio-42/elFinder#1029]
Hi,
This script is very useful, but it's too buggy!
How can we make the plugin to ask for file name when creating archives?
Also, how can we extract archives to other directory than the current?

A bug that I already fixed in my own, When we edit files, it's better to keep the dialog open, we do save and continue editting, it's better to give open for editors to control whether to exit after save or not, I had to edit the edit.js file to remove the deferred object and the close call after the save method. See example of my edits in attached image
![localhost_filemanager](https://cloud.githubusercontent.com/assets/6576868/8098248/5a02fb6c-0fe5-11e5-995b-2a55a655e894.png)

Thanks

**Comment by nao-pon** [comment]
@ERROPiX thanks! :+1:

**Comment by ERROPiX** [comment]
Thank you for updating the files

**Comment by megadr01d** [comment]
@ERROPiX Sorry to bother, is that CodeMirror integration?

**Comment by ERROPiX** [comment]
No, it's ace integration

**Comment by megadr01d** [comment]
I actually prefer Ace, it's what Codiad uses. Is it easy to integrate? If it's simple and you find the time to share the relevant config, drop me a line on megadroid [at] microvoid [dot] in. Otherwise, thanks anyway for the info.

**Comment by ERROPiX** [comment]
It's not really hard to do.
I will share the config with you here or maybe add it wiki later if @nao-pon accept!

my setup of elFinder is quit tricky, I have put all frontend files in my local server, and combined all connector classes in one file that I upload to my clients servers to manage files with only one file.

This is my elFinder configuration:

```html
<!DOCTYPE html>
<html>
	<head>
		...
		<script type="text/javascript" src="js/ace/ace.js"></script>
		<script type="text/javascript">
		$(window).load(function() {
			$("#elfinder").elfinder({
				commandsOptions : {
					edit : {
						// edit files in wysisyg's
						editors : [
							{
								mimes : [
									'text/html',
									'text/xml',
									'text/javascript',
									'text/json',
									'text/css',
									'text/x-php'
								],
							 
								load : function(textarea, file) {
									console.debug(fm);
									
									$(textarea).hide();
									$(textarea)
										.closest('.ui-dialog')
											.css('width','1010px')
											.addClass('ace-dialog');
									
									$('<div id="ace-code-editor"></div>')
									 .insertAfter(textarea)
									 .text( textarea.value );
									
									var editor = ace.edit( "ace-code-editor" );
									editor.setTheme("ace/theme/xcode");
									
									var mode = '';
									switch( file.mime ) {
										case 'text/html':
											mode = 'HTML'
											break;
										case 'text/xml':
											mode = 'XML'
											break;
										case 'text/css':
											mode = 'CSS'
											break;
										case 'text/javascript':
											mode = 'JavaScript'
											break;
										case 'text/json':
											mode = 'JSON'
											break;
										case 'text/x-php':
											mode = 'PHP'
											break;
									}
									
									if( mode ) {
										editor.getSession().setMode("ace/mode/"+mode.toLowerCase());
										$("#elfinder .elfinder-dialog .ui-dialog-buttonpane")
										.prepend('<span>Mode '+mode+'</span>');
									}
									return editor;
								},
								
								close : function(textarea, editor) {},
								
								save : function(textarea, editor) {
									textarea.value = editor.getValue();
								}
							}
						]
					},
				}
			});
		});
		</script>
		<!-- elfinder Custom CSS -->
		<style type="text/css">
		/* custom */
		body {
			margin: 0;
			overflow: hidden;
		}
		#elfinder,
		#elfinder > .elfinder-toolbar,
		#elfinder > .elfinder-statusbar {
			border-radius: 0;
		}
		#elfinder .elfinder-cwd table {
			margin: 0;
		}
		#elfinder .elfinder-dialog {
			top: 50% !important;
			left: 50% !important;
			transform: translate(-50%,-50%) !important;
		}
		#elfinder .ace-dialog {
			box-shadow: 0 0 1000px rgba(0,0,0, .7);
		}
		#elfinder .ace-dialog .ui-dialog-buttonpane > span {
			line-height: 25px;
			display: inline-block;
		}
		#elfinder .ace-dialog .ui-resizable-handle {
			display: none;
		}
		#elfinder .elfinder-dialog-notify {
			z-index: 100 !important;
		}
		#ace-code-editor {
			margin: 0;
			position: relative;
			height: 600px;
			width: 1000px;
			font-size: 16px;
		}
		</style>
	</head>
	<body>
		<div id="elfinder"></div>
	</body>
</html>
```html

**Comment by megadr01d** [comment]
> maybe add it wiki later if @nao-pon accept!

That would be a valuable contribution to the HOWTOs.
Thanks a lot, man. I'm gonna try it out and let you know the results.

**Comment by megadr01d** [comment]
@ERROPiX In v2.1 nightly, I got it working with a few modifications. I had to remove `console.debug(fm);` and change the `editor` var to something like `this.AceEditor` (so the save buttons would work). I always get some kind of error in the console with `switch( file.mime )` so I couldn't get the mode to change dynamically according to the MIME type. Everything else worked OK.

Anyway, I found CodeMirror is easier to integrate and I'm happy with it. Thanks again for the useful pointers to integrate both editors.
